File: app/src/main/java/org/bochman/upapp/domain/interactor/SearchNearby.java
package org.bochman.upapp.domain.interactor;

import com.google.android.gms.maps.model.LatLng;

import org.bochman.upapp.data.repository.PoiRepository;

/**
 * Encapsulate a nearby search request.
 */
public class SearchNearby {

    private final PoiRepository repo;

    SearchNearby(PoiRepository repo){
        this.repo=repo;
    }

    void invoke(LatLng latlng){
        repo.fetchPoiByLatLng(latlng);
        repo.saveSearchQuery("");
    }

}

File: app/src/main/java/org/bochman/upapp/domain/interactor/SearchNearby.kt
package org.bochman.upapp.domain.interactor;

import androidx.lifecycle.LiveData
import com.google.android.gms.maps.model.LatLng;
import org.bochman.upapp.data.enteties.Poi

import org.bochman.upapp.data.repository.PoiRepository;

/**
 * Encapsulate a nearby search request.
 */
class SearchNearby(private val poiRepository: PoiRepository){

        suspend operator fun invoke(latlng : LatLng) =  poiRepository.fetchPoiByLatLng(latlng)
}
